### PR TITLE
Icons: Deprecate `warning` and rename to `cautionFilled`

### PR DESCRIPTION
**Pull request by mirka**
Follow-up to https://github.com/WordPress/gutenberg/pull/66555/files#r1880778506

## What?

Deprecates the `warning` icon and renames (re-exports) it as `cautionFilled`.

## Why?

The Icons package has a convention where we have a pair of filled and unfilled icons:

Help icon pair

The `warning` icon is a filled icon, but didn't follow the naming convention:

Warning icon

We now have a need for a corresponding "unfilled" icon, and thus should take this opportunity to streamline the naming.

Caution icon, unfilled

## Testing Instructions

- The `warning` icon should appear in the [Storybook Icon Library](http://localhost:50240/?path=/story/icons-icon--library) with the new name `cautionFilled`.
- The icon can still be imported with the name `warning`, but marked as deprecated in TypeScript.

There are no existing usages of the `warning` icon in the app (as far as my regex skills tells me), but maybe you can find some!

## Screenshots or screencast 

|Before|After|
|-|-|
| Warning/caution icon before | Warning/caution icon after |

### CODE REVIEW

File: packages/icons/src/icon/stories/index.story.js [hunk 1]
```diff
@@ -11,7 +11,14 @@ import check from '../../library/check';
 import * as icons from '../../';
 import keywords from './keywords';
 
-const { Icon: _Icon, ...availableIcons } = icons;
+const {
+	Icon: _Icon,
+
+	// Deprecated aliases
+	warning: _warning,
```
Comment (mirka) [inline packages/icons/src/icon/stories/index.story.js:18]: Omitting from the Icon Library in Storybook.

File: packages/icons/src/icon/stories/keywords.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 const keywords: Partial< Record< keyof typeof import('../../'), string[] > > = {
 	cancelCircleFilled: [ 'close' ],
+	cautionFilled: [ 'alert', 'caution', 'warning' ],
```
Comment (mirka) [inline packages/icons/src/icon/stories/keywords.ts:3]: Making sure it's still findable with the search query "warning".

**Review by jameskoster — APPROVED** [review]
Works for me :)

**Review by tyxla — APPROVED** [review]
Tests well and code LGTM 👍 

I've just updated a couple of RN test snapshots that were still expecting the old icon.

**Comment by mirka** [comment]
Oops, I was omitting native files from my search. Fixed in 7f1d0ac4ee8e4fc2d6d2e3fa5cb15ac5ccd04145.

**Pull request merged**